modeshape-jcr/src/main/java/org/modeshape/jcr/cache/document/WorkspaceCache.java

```diff
@@ -249,10 +248,8 @@ public CachedNode getNode( NodeKey key ) {
                 CachedNode newNode = new LazyCachedNode(key, doc);
                 try {
                     Integer cacheTtlSeconds = translator().getCacheTtlSeconds(doc);
-                    if (cacheTtlSeconds != null && nodesByKey instanceof org.infinispan.commons.api.BasicCache) {
-                        node = ((org.infinispan.commons.api.BasicCache<NodeKey, CachedNode>)nodesByKey).putIfAbsent(key, newNode,
-                                                                                         cacheTtlSeconds.longValue(),
-                                                                                         TimeUnit.SECONDS);
+                    if (cacheTtlSeconds != null) {
+                        //TODO author=Horia Chiorean date=10/11/2015 description=Cache TTL seconds are not supported by the current cache
```

Comment:
Okay, I've read your description in https://issues.jboss.org/browse/MODE-2529 and understand the TODO.